src/service/viewer-impl.js

```diff
@@ -463,7 +477,50 @@ export class Viewer {
    * @return {!VisibilityState}
    */
   getVisibilityState() {
-    return this.visibilityState_;
+    const state = this.visibilityState_;
```

Comment:
Hmm. I feel rather strongly that we should do this all in the setter. We should know what the state is before telling everyone that it's changed. Seems like it should go fairly easily into setter, no?